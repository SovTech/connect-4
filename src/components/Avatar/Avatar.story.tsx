import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { TEST_AVATAR_URL } from '../../constants/dummy_content';
import { Avatar, Card } from '../index';

storiesOf('Avatar', module)
  .add('User with image', () => (
    <Card>
      <Avatar
        imageUrl={TEST_AVATAR_URL}
        size={50}
      />
    </Card>
  ))
  .add('User without image', () => (
    <Card>
      <Avatar
        size={50}
      />
    </Card>
  ))
  .add('User without image - small', () => (
    <Card>
      <Avatar
        size={30}
      />
    </Card>
  ))
  .add('Non user without image - small', () => (
    <Card>
      <Avatar
        size={30}
      />
    </Card>
  ));
