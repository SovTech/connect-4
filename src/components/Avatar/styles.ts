import styled from 'styled-components';
import theme from '../../theme';

type WrapperProps = {
  size: 30 | 40 | 50 | 80;
}

type InnerProps = {
  size: 30 | 40 | 50 | 80;
}

export const AvatarWrapper = styled<WrapperProps, 'div'>('div')`
  background-color: ${theme.overlayColor};
  border-radius: 50%;
  margin: 2px;
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  
    &.YELLOW {
      border: 4px solid ${theme.warning};
    }
    
    &.RED {
      border: 4px solid ${theme.error};
    }
`;

export const AvatarInner = styled<InnerProps, 'div'>('div')`
  border-radius: 50%;
  background-size: contain;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
`;
