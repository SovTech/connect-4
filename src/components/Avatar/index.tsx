// User and Client Avatar to keep them all consistent

import * as React from 'react';
import { AvatarInner, AvatarWrapper } from './styles';

type Props = {
  imageUrl?: string;
  size: 30 | 40 | 50 | 80;
  activeColor?: 'RED' | 'YELLOW';
}

const Avatar = (props: Props) => {
  return (
    <AvatarWrapper size={props.size} className={props.activeColor}>
      {props.imageUrl
        ? <AvatarInner
          size={props.size}
          style={{backgroundImage: `url("${props.imageUrl}")`}}
        />
        : null}
    </AvatarWrapper>
  );
};

export default Avatar;
