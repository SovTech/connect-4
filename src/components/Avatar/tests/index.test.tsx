import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Avatar from '../index';

describe('Avatar Component', () => {
  it('renders a user Avatar correctly when there is a url', () => {
    const tree = renderer
      .create(<Avatar imageUrl='https://dummyimage.png' size={50} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a user Avatar correctly when there is no url', () => {
    const tree = renderer
      .create(<Avatar size={50} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a non-user Avatar correctly when there is a url', () => {
    const tree = renderer
      .create(<Avatar imageUrl='https://dummyimage.png' size={50} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a non-user Avatar correctly when there is no url', () => {
    const tree = renderer
      .create(<Avatar size={50} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a small Avatar correctly', () => {
    const tree = renderer
      .create(<Avatar imageUrl='https://dummyimage.png' size={30} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
