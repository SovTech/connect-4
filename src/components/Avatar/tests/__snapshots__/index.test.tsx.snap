// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component renders a non-user Avatar correctly when there is a url 1`] = `
<div
  className="sc-bdVaJa fwEAqZ"
  size={50}
>
  <div
    className="sc-bwzfXH jeCDFy"
    size={50}
    style={
      Object {
        "backgroundImage": "url(\\"https://dummyimage.png\\")",
      }
    }
  />
</div>
`;

exports[`Avatar Component renders a non-user Avatar correctly when there is no url 1`] = `
<div
  className="sc-bdVaJa fwEAqZ"
  size={50}
/>
`;

exports[`Avatar Component renders a small Avatar correctly 1`] = `
<div
  className="sc-bdVaJa hspnGX"
  size={30}
>
  <div
    className="sc-bwzfXH fFqIba"
    size={30}
    style={
      Object {
        "backgroundImage": "url(\\"https://dummyimage.png\\")",
      }
    }
  />
</div>
`;

exports[`Avatar Component renders a user Avatar correctly when there is a url 1`] = `
<div
  className="sc-bdVaJa fwEAqZ"
  size={50}
>
  <div
    className="sc-bwzfXH jeCDFy"
    size={50}
    style={
      Object {
        "backgroundImage": "url(\\"https://dummyimage.png\\")",
      }
    }
  />
</div>
`;

exports[`Avatar Component renders a user Avatar correctly when there is no url 1`] = `
<div
  className="sc-bdVaJa fwEAqZ"
  size={50}
/>
`;
